sky/skypilot_config.py

```diff
@@ -116,9 +117,10 @@
 _GLOBAL_CONFIG_PATH = '~/.sky/config.yaml'
 _PROJECT_CONFIG_PATH = '.sky.yaml'
 
-_SQLALCHEMY_ENGINE: Optional[sqlalchemy.engine.Engine] = None
 API_SERVER_CONFIG_KEY = 'api_server_config'
 
+_DB_USE_LOCK = threading.Lock()
```

Comment:
Just to be safe, lets use a reentrant lock, shouldn't matter.
```suggestion
_DB_USE_LOCK = threading.RLock()
```